maint/docnotes

```diff
@@ -151,13 +151,21 @@ N*/
 
 -----------------------------------------------------------------------------
 MPI-2 deprecated some routines (see 2.6.1)
-/*N Deprecated
+/*N DeprecatedMPI2
```

Comment:
Moving forward, the new release of MPICH will assume MPI-4, so "deprecated"  means it is deprecated for the current release, or MPI-4.0 in this case. I think simply modify the doc text removing the mention of MPI-2 or MPI-3 is good enough.

Reply:
Thanks. I fixed it.